migration-engine/migration-engine-tests/tests/migrations/basic/vitess.rs

```diff
@@ -0,0 +1,54 @@
+use migration_engine_tests::test_api::*;
+
+#[test_connector(tags(Vitess), preview_features("referentialIntegrity"))]
+fn reordering_and_altering_models_at_the_same_time_works(api: TestApi) {
+    let dm1 = r#"
+        model A {
+            id Int @id
+            name Int @unique
+            c C @relation(name: "atoc", fields: [name], references: [name], onUpdate: Restrict)
+            cs C[] @relation(name: "ctoa")
+        }
+
+        model B {
+            id Int @id
+            name Int @unique
+            c C @relation(name: "btoc", fields: [name], references: [name], onUpdate: Restrict)
+        }
+
+        model C {
+            id Int @id
+            name Int @unique
+            a A @relation(name: "ctoa", fields: [name], references: [name], onUpdate: Restrict)
+            as A[] @relation(name: "atoc")
+            bs B[] @relation(name: "btoc")
+        }
+    "#;
+
+    api.schema_push_w_datasource(dm1).send().assert_green();
+
+    let dm2 = r#"
+        model C {
+            id Int @id
+            a A @relation(name: "ctoa2", fields: [name], references: [name], onUpdate: Restrict)
+            name Int @unique
+            bs B[] @relation(name: "btoc2")
+            as A[] @relation(name: "atoc2")
+        }
+
+        model A {
+            id Int @id
+            name Int @unique
+            c C @relation(name: "atoc2", fields: [name], references: [name], onUpdate: Restrict)
+            cs C[] @relation(name: "ctoa2")
+        }
+
+        model B {
+            c C @relation(name: "btoc2", fields: [name], references: [name], onUpdate: Restrict)
+            name Int @unique
+            id Int @id
+        }
+    "#;
+
+    api.schema_push_w_datasource(dm2).send().assert_green();
```

Comment:
I don't understand why this one doesn't work out of the box

Reply:
Because only `Restrict` and `SetNull` are allowed if integrity is handled by Prisma. This could also be `Restrict` for every other database and a separate test with `NoAction` for SQL Server.